functions/Test-DbaSqlBuild.ps1

```diff
@@ -0,0 +1,246 @@
+        License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+    .LINK
+        https://dbatools.io/Test-DbaSqlBuild
+#>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    [CmdletBinding()]
+    param (
+        [version[]]
+        $Build,
+
+        [version]
+        $MinimumBuild,
+
+        [string]
+        $MaxBehind,
+
+        [parameter(Mandatory = $false, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]
+        $SqlInstance,
+
+        [Alias("Credential")]
+        [PsCredential]
+        $SqlCredential,
+
+        [switch]
+        $Update,
+
+        [switch]
+        [Alias('Silent')]$EnableException
+    )
+
+    begin {
+        #region Helper functions
+        function Get-DbaSqlBuildReferenceIndex {
+            [CmdletBinding()]
+
+            $DbatoolsData = Get-DbaConfigValue -Name 'Path.DbatoolsData'
+            $writable_idxfile = Join-Path $DbatoolsData "dbatools-buildref-index.json"
+            $result = Get-Content $writable_idxfile -Raw | ConvertFrom-Json
+            $result.Data | Select-Object @{ Name = "VersionObject"; Expression = { [version]$_.Version } }, *
+        }
+        if ($MaxBehind) {
+            $MaxBehindValidator = [regex]'^(?<howmany>[\d]+)(?<what>SP|CU)$'
+            $pieces = $MaxBehind.Split(' ')	| Where-Object { $_ }
+            try {
+                $ParsedMaxBehind = @{}
+                foreach ($piece in $pieces) {
+                    $pieceMatch = $MaxBehindValidator.Match($piece)
+                    if ($pieceMatch.Success -ne $true) {
+                        Stop-Function -Message "MaxBehind has an invalid syntax ('$piece' could not be parsed correctly)" -ErrorRecord $_
+                        return
+                    }
+                    else {
+                        $howmany = [int]$pieceMatch.Groups['howmany'].Value
+                        $what = $pieceMatch.Groups['what'].Value
+                        if ($ParsedMaxBehind.ContainsKey($what)) {
+                            Stop-Function -Message "The specifier $what has been already passed" -ErrorRecord $_
+                            return
+                        }
+                        else {
+                            $ParsedMaxBehind[$what] = $howmany
+                        }
+                    }
+                }
+                if (-not $ParsedMaxBehind.ContainsKey('SP')) {
+                    $ParsedMaxBehind['SP'] = 0
+                }
+            }
+            catch {
+                Stop-Function -Message "Error parsing MaxBehind" -ErrorRecord $_
+                return
+            }
+        }
+    }
+    process {
+        if (Test-FunctionInterrupt) { return }
+        try {
+            $IdxRef = Get-DbaSqlBuildReferenceIndex -Moduledirectory $moduledirectory -Update $Update -EnableException $EnableException
```

Comment:
This looks like something that should run in `begin`
The parameters aren't on the helper function defined in begin.
When calling one of our commands in try/catch, you shouldn't pass through `-EnableException $EnableException`, rather set it to `$true`. Also keep in mind that it is a switch parameter.

Reply:
blunt copy/paste mistake. Thanks for the catch